entity-resolution/java/src/main/java/org/linqs/psl/examples/entityresolution/Run.java

```diff
@@ -0,0 +1,306 @@
+            for (Constant argument : atom.getArguments()) {
+                writer.write(argument.toString() + "\t");
+            }
+            writer.write("" + atom.getValue() + "\n");
+        }
+
+        writer.close();
+        resultsDB.close();
+    }   
+
+    /**
+     * Run statistical evaluation scripts to determine the quality of the inferences
+     * relative to the defined truth.
+     * Note that the target predicate is categorical and we will assign the category with the
+     * highest truth value as true and the rest false.
+     */
+    private void evalResults() {
+        Set<StandardPredicate> closedPredicates = new HashSet<StandardPredicate>();
+        closedPredicates.add(model.getStandardPredicate("AuthorName"));
+        closedPredicates.add(model.getStandardPredicate("AuthorOf"));
+        closedPredicates.add(model.getStandardPredicate("PaperTitle"));
+
+        Set<StandardPredicate> openPredicates = new HashSet<StandardPredicate>();
+        openPredicates.add(model.getStandardPredicate("SameAuthor"));
+        openPredicates.add(model.getStandardPredicate("SamePaper"));
+
+        Partition targetPartition = dataStore.getPartition(PARTITION_EVAL_TARGETS);
+        Partition observationsPartition = dataStore.getPartition(PARTITION_EVAL_OBSERVATIONS);
+        Partition truthPartition = dataStore.getPartition(PARTITION_EVAL_TRUTH);
+
+        // Because the truth data also includes observed data, we will make sure to include the observed
+        // partition here.
+        Database resultsDB = dataStore.getDatabase(targetPartition, closedPredicates, observationsPartition);
+        Database truthDB = dataStore.getDatabase(truthPartition, dataStore.getRegisteredPredicates());
+
+        Evaluator eval = new DiscreteEvaluator();
+
+        for (StandardPredicate targetPredicate : openPredicates) {
+            eval.compute(resultsDB, truthDB, targetPredicate);
+            log.info(eval.getAllStats());
+        }
+
+        resultsDB.close();
+        truthDB.close();
+    }
+
+    public void run() {
+        definePredicates();
+        defineRules();
+        loadData();
+
+        learnWeights();
+        runInference();
+
+        try{
+            writeOutput();
+        } catch (IOException ex){
+            throw new RuntimeException("Unable to write out results.", ex);
+        }
+        evalResults();
+
+        dataStore.close();
+    }
+
+    /**
+     * Run this model from the command line.
+     */
+    public static void main(String[] args) {
+        Run run = new Run();
+        run.run();
+    }
+
+    private static String getHostname() {
+        String hostname = "unknown";
+
+        try {
+            hostname = InetAddress.getLocalHost().getHostName();
+        } catch (UnknownHostException ex) {
+            log.warn("Hostname can not be resolved, using '" + hostname + "'.");
+        }
+        return hostname;
+    }
+}
```

Comment:
Missing a newline.
(Not an empty line, just a newline.)
(Ask me if you are still confused.)